src/module.c

```diff
@@ -6704,6 +6733,115 @@ int RM_SetModuleUserACL(RedisModuleUser *user, const char* acl) {
     return ACLSetUser(user->user, acl, -1);
 }
 
+/* Creates a user ID from the user created through the redis module interface.
+ * The user ID can be used later, in order to check if command, key or channel
+ * can be executed or accessed according to the ACLs rules associated with that user.
+ * The caller can later free the user ID using the function RM_FreeModuleUserID
+ *
+ * On success a RedisModuleUserID object is returned, otherwise
+ * NULL is returned and errno is set to ENOENT. */
+ RedisModuleUserID *RM_CreateModuleUserID(RedisModuleCtx *ctx) {
+    if (ctx->client->user == NULL) {
+        errno = ENOENT;
```

Comment:
i don't think we wanna mess with errno in most APIs, just when we have multiple failure reasons.
```suggestion
```